org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -342,71 +299,55 @@ private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource, Analys
     if (iTextFileBufferManager == null) {
       return;
     }
+
     Set<IFile> failedFiles = result.failedAnalysisFiles().stream().map(f -> f.<IFile>getClientObject()).collect(Collectors.toSet());
-    PreviousMarkerCache markerCache = new PreviousMarkerCache(this.request, failedFiles);
-    long now = new Date().getTime();
     for (Entry<IResource, List<Issue>> resourceEntry : issuesPerResource.entrySet()) {
-      IResource r = resourceEntry.getKey();
-      if (failedFiles.contains(r)) {
+      IResource resource = resourceEntry.getKey();
+      if (failedFiles.contains(resource)) {
         continue;
       }
-      List<IMarker> previousMarkers = markerCache.getPrevious(r);
-      Object lastAnalysis = r.getSessionProperty(LAST_ANALYSIS_PROP_NAME);
-      Long creationTimeStampForNewIssues;
-      if (previousMarkers.isEmpty() && lastAnalysis == null) {
-        creationTimeStampForNewIssues = null;
-      } else {
-        creationTimeStampForNewIssues = now;
-      }
       List<Issue> rawIssues = resourceEntry.getValue();
       try {
-        if (r instanceof IFile) {
-          issueTrackingOnFile(iTextFileBufferManager, r, previousMarkers, rawIssues, creationTimeStampForNewIssues);
+        if (resource instanceof IFile) {
+          IProject project = resource.getProject();
+          String moduleKey = project.getName();
+          SonarLintCorePlugin.getDefault().getModulePathManager().setModulePath(moduleKey, project.getLocation().toString());
+
+          String relativePath = resource.getProjectRelativePath().toString();
+          trackLocalIssues(moduleKey, resource, relativePath, rawIssues);
+          trackServerIssues(moduleKey, resource, relativePath);
         } else {
-          issueTracking(r, previousMarkers, rawIssues, null, creationTimeStampForNewIssues);
+          // TODO delete if this never happens, or else handle it better
```

Comment:
I think this part was to support issues reported at project level (for example rule about missing package-info.java)